### DISCUSSION TITLE
Sniffing the serial com between motor main board and his LC12s module

### DISCUSSION THREAD

**Discussion by meute** [Yogui79/IntexPureSpa#54] [Ideas]
Hi,

Someone have try to snif directly the wired serial communication on the LC12s on the motor main board ?
Particularly at motor power on ...

There is big probability that the motor start the communication on a specific channel with a specific network ID who are always the sames at power up until the remote controller can catch it and start to communicate to get the final channel and network ID that the motor want to pair.

And when the remote controller confirm to the motor that she have received the final settings both are switching to this new settings to continue to exchange.

It's the only way I see to achieve an "auto pairing".

And the remote controller probably fallback to these constant channel and ID when she lose the communication for some time, in case the motor is powered off for example.

I made pure guesses, but by sniffing the wired serial exchange it might be possible to confirm it or not and after that maybe be able to reverse engineer the pairing process.

**Comment by LEECHER1** [comment]
hey,

We did it the same way at the beginning and we have the same assumption that the pump and the controller check the channels in advance, but unfortunately we couldn't read out any exact data so that we could get an auto pairing together.

But @Yogui79 can explain better.

At home I have a pump circuit board and a controller to check/read.

**Comment by meute** [comment]
Hi @LEECHER1 

I have just opened my motor and connected an arduino mega on RX and SET pin of the integrated LS12s (Blue board version, not red) and started to sniffing all the messages and SET signal change but without success for now.

I can snif all the messages between the motor and the remote controller (with that I can get the channel, thank to the first char in each message but obviously not the network ID).

But I see nothing about a configuration message from the main board to the LS12s, it's like if the motor never try to setup the LS12s.
Also if I make the remote controller unreachable, the motor not seems to fall back on some sort of "initialization mode" with special LS12s settings.

Maybe this LS12s have a custom firmware from Intex who handle itself the pairing ... I don't know.

**Reply by meute** [reply]
I soldered a 3 pin M8 industrial sensor connector which I can reach by opening the ground connection box, so I can now connect or disconnect the sniffer whenever I want without having to reopen the motor.
I then re-tropicalised with silicone the area where I soldered the wires

![image](https://user-images.githubusercontent.com/35557846/175364817-2dd28e35-bf81-4eeb-a57d-af2d9150fd41.png)

![image](https://user-images.githubusercontent.com/35557846/175363934-fe8faf9d-90c2-434a-9a89-347a9d662373.png)

**Comment by redeye86** [comment]
Hi, before i found this project I used the included controller unit of our intex 28462. It was the older model from 2021 according to the serial number. This one is used in conjunction with the intex app. Because this one temporarely stopped working and also it was very unreliable as far as the wifi was concerned, I bought the newer model that uses the tuya protocol / app for the smart home connection. In theory, this is a bit more reliable regarding wifi, but the home assistant intergration relies on the tuya protocol which also did not work well for me. So I am also looking for a solution.

Interestingly, the pairing between the control unit and the pump happens automatically as soon as you place the control unit in the tray on the pump. Inductive charging takes place there. Perhaps inductive charging simply triggers a pairing mode? AFAIK, the charging coil does not include any data connection or similar, as there is only one pair of wires. 

On the other hand, my attempt to charge the controller outside the pump with a replacement coil I had ordered was strangely unsuccessful. My plan was to have the control unit in a 3d printed stand in the house and charge it there for better Wi-Fi reception and less splashing water and sun exposure. But although I powered that coil with the same voltage the control unit was not indicating that it is loading. So maybe also for the loading there could be another condition that was not met.

Also at the bottom of the loading tray, there is some round part that could be some kind of lid. Maybe there is a rfid chip that holds the channel and network id?
